opentelemetry-sdk/tests/logs/test_handler.py

```diff
@@ -86,105 +88,114 @@ def test_log_flush_noop(self):
+                {**log_record.attributes, "http.status_code": 200},
+            )
+            self.assertTrue(
+                log_record.attributes[SpanAttributes.CODE_FILEPATH].endswith(
+                    "test_handler.py"
+                )
+            )
+            self.assertEqual(
+                log_record.attributes[SpanAttributes.CODE_FUNCTION],
+                "test_log_record_user_attributes",
+            )
+            # The line of the log statement is not a constant (changing tests may change that),
+            # so only check that the attribute is present.
+            self.assertTrue(
+                SpanAttributes.CODE_LINENO in log_record.attributes
+            )
+            self.assertTrue(
+                isinstance(log_record.attributes, BoundedAttributes)
             )
-        )
-        self.assertEqual(
-            log_record.attributes[SpanAttributes.CODE_FUNCTION],
-            "test_log_record_user_attributes",
-        )
-        # The line of the log statement is not a constant (changing tests may change that),
-        # so only check that the attribute is present.
-        self.assertTrue(SpanAttributes.CODE_LINENO in log_record.attributes)
-        self.assertTrue(isinstance(log_record.attributes, BoundedAttributes))
 
     def test_log_record_exception(self):
         """Exception information will be included in attributes"""
-        processor, logger = set_up_test_logging(logging.ERROR)
-
-        try:
-            raise ZeroDivisionError("division by zero")
-        except ZeroDivisionError:
-            with self.assertLogs(level=logging.ERROR):
-                logger.exception("Zero Division Error")
+        with set_up_test_logging(logging.ERROR) as (processor, logger):
+            try:
+                raise ZeroDivisionError("division by zero")
+            except ZeroDivisionError:
+                with self.assertLogs(level=logging.ERROR):
+                    logger.exception("Zero Division Error")
 
-        log_record = processor.get_log_record(0)
+            log_record = processor.get_log_record(0)
 
-        self.assertIsNotNone(log_record)
-        self.assertEqual(log_record.body, "Zero Division Error")
-        self.assertEqual(
-            log_record.attributes[SpanAttributes.EXCEPTION_TYPE],
-            ZeroDivisionError.__name__,
-        )
-        self.assertEqual(
-            log_record.attributes[SpanAttributes.EXCEPTION_MESSAGE],
-            "division by zero",
-        )
-        stack_trace = log_record.attributes[
-            SpanAttributes.EXCEPTION_STACKTRACE
-        ]
-        self.assertIsInstance(stack_trace, str)
-        self.assertTrue("Traceback" in stack_trace)
-        self.assertTrue("ZeroDivisionError" in stack_trace)
-        self.assertTrue("division by zero" in stack_trace)
-        self.assertTrue(__file__ in stack_trace)
+            self.assertIsNotNone(log_record)
+            self.assertIn("Zero Division Error", log_record.body)
```

Comment:
This hides a bad breaking change. Exception messages now weirdly add a newline and NoneType parameter.